### ISSUE TITLE
simplify canary / canaryd configuration

### ISSUE THREAD

**Issue by gorsuch** [canaryio/canary#44]
For the sake of simplifying the codebase, I'm interested in exploring the following:

* moving to a `flags` based setup for both `canary` and `canaryd`
* pushing most of `canaryd`'s config to the manifest

Before I can do that, though, I'd like to get some feedback from how others are using `canary` / `canaryd` today. (cc @jmeichle @blalor and anyone else who wants to comment).  I don't want to ruin anyone's day, but I def want to make config easier to reason about.

**Comment by jmeichle** [comment]
I agree completely with moving to a flags based setup for both binaries. I would also agree that having more configuration in the manifest for `canaryd` is preferable for many reasons. 

Currently I am not utilizing `canaryd` for anything beyond research purposes, but hope to start using it for a large site monitoring project soon (pending other projects and resource allocation). There is no worry about changing behavior/configuration for at least myself currently.
